### PR TITLE
Animal Spawning and Respawning

### PR DESCRIPTION
**Pull request by JaredTrenholm**

### CODE REVIEW

**Review by 2401lucas — CHANGES_REQUESTED** [review]
AnimalManager: 
Remove Update and put coroutine in place (With 1 second delay approx)
Have a bool to use respawning
No Overlap sphere, Use hashtable instead to save relation of checkpoint/respawn timer
Remove InitialSpawn
One Animal, set the mesh filter and renderer to the animal you want, tell the animal what kind of animal it is 
Start with only small animals, then some medium, then some large, then no small, then no medium. Animal choice is based on random chance IE 75% for small, 20% for medium, 5% Large

File: Assets/Scripts/Managers/AnimalManager.cs [hunk 1]
```diff
@@ -6,94 +7,166 @@ namespace AlienGame.Managers
 {
     public class AnimalManager : MonoBehaviour
     {
+        #region Fields
         [Header("Settings")]
         public bool debug = false;
+        public bool canRespawn = false;
```
Comment (2401lucas) [inline Assets/Scripts/Managers/AnimalManager.cs:13]: Should be useRespawn

File: Assets/Scripts/Managers/AnimalManager.cs [hunk 2]
```diff
@@ -6,94 +7,166 @@ namespace AlienGame.Managers
...
+        public void EndManager()
         {
-            started = false;
-            timePassed = 0f;
-            animalCount = 0;
+            StopCoroutine(Respawn());
+            StopCoroutine(ChangeWave());
         }
-        public void StartSpawn()
+        public void StartManager()
```
Comment (2401lucas) [inline Assets/Scripts/Managers/AnimalManager.cs:47]: Should Also clear the Job table

File: Assets/Scripts/Managers/AnimalManager.cs [hunk 3]
```diff
@@ -6,94 +7,166 @@ namespace AlienGame.Managers
...
+        public Material mediumAnimalMaterial;
+        public Material largeAnimalMaterial;
 
         private GameObject[] animalSpawnPoints;
-        private float timePassed = 0f;
-        private bool started = false;
-        private int animalCount;
-        private float animalSize = 3f;
+        private WaitForSeconds respawnDelay;
+        private Hashtable animalSpawns = new Hashtable();
```
Comment (2401lucas) [inline Assets/Scripts/Managers/AnimalManager.cs:27]: Hashtable should contain all checkpoints and respawn timers on checkpoints with no animals. If the checkpoint has an actively spawned animal, we do not want it in the hashtable.

File: Assets/Scripts/Managers/AnimalManager.cs [hunk 4]
```diff
@@ -6,94 +7,166 @@ namespace AlienGame.Managers
...
+                {
+                    Log("Attempt Spawn");
+                    Spawn();
+                }
+                yield return respawnDelay;
             }
         }
 
-        private void RespawnCheck()
+        private bool CanSpawn()
```
Comment (2401lucas) [inline Assets/Scripts/Managers/AnimalManager.cs:77]: Can spawn should not contain an if statement, it should simply return  the condition

File: Assets/Scripts/Managers/AnimalManager.cs [hunk 5]
```diff
@@ -6,94 +7,166 @@ namespace AlienGame.Managers
...
+        }
+        #endregion
+
+        #region Set Animals and Waves
+
+        //TODO
+        //Set up a method on animals to tell it what animal it is
+        private void SetAnimal(GameObject animal)
+        {
+            float animalResult = UnityEngine.Random.Range(0, 101);
```
Comment (2401lucas) [inline Assets/Scripts/Managers/AnimalManager.cs:129]: The upper value of range should be the sum of the percentages of the animal spawn chances

**Review by 2401lucas — CHANGES_REQUESTED** [review]

**Review by SirLorrence — APPROVED** [review]
Everything seems to be in order.

**Pull request merged**